# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/powerpc/pdisk/linux_prio_tree_h.patch
# ROCK Linux is Copyright (C) 1998 - 2005 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- ./file_media.c.orig	2004-11-26 17:40:27.315376952 +0100
+++ ./file_media.c	2004-11-26 17:41:09.681936256 +0100
@@ -40,6 +40,11 @@
 
 #ifdef __linux__
 #include <sys/ioctl.h>
+/* for linux-2.6.9, from linux/types.h */
+#  include <asm/types.h>
+#  ifndef pgoff_t
+#  define pgoff_t unsigned long
+#  endif
 #include <linux/fs.h>
 #include <linux/hdreg.h>
 #include <sys/stat.h>
--- ./pdisk.c.orig	2004-11-26 17:39:02.005346048 +0100
+++ ./pdisk.c	2004-11-26 17:39:53.121575200 +0100
@@ -51,6 +51,11 @@
 
 #ifdef __linux__
 #include <sys/ioctl.h>
+/* for linux-2.6.9, from linux/types.h */
+#  include <asm/types.h>
+#  ifndef pgoff_t
+#  define pgoff_t unsigned long
+#  endif
 #include <linux/fs.h>
 #include <linux/hdreg.h>
 #endif
